pnm-plugin (#4253)
AcademySoftwareFoundation/OpenImageIO: src/doc/builtinplugins.rst

PNM / Netpbm
===============================================

The Netpbm project, a.k.a. PNM (portable "any" map) defines PBM, PGM, PPM
and later added PFM (portable float map) as a set of simple image formats
(portable bitmap, portable graymap, portable pixmap) files.
Without loss of generality, we will refer to these all collectively as
"PNM."  These files have extensions :file:`.pbm`, :file:`.pgm`,
:file:`.ppm`, :file:`.pfm` and customarily correspond to bi-level bitmaps,
1-channel grayscale, and 3-channel RGB files, respectively, or :file:`.pnm`
for those who reject the nonsense about naming the files depending on the
number of channels and bitdepth.

> PNM files widely used in the Unix world as simple ASCII or binary image 

Reviewer: ```suggestion
PNM files are widely used in the Unix world as simple ASCII or binary image 
```

     - int
     - The true bits per sample of the file (1 for true PBM files, even
       though OIIO will report the ``format`` as UINT8).
   * - ``pnm:binary``
     - int
     - nonzero if the file itself used the PNM binary format, 0 if it used
       ASCII.  The PNM writer honors this attribute in the ImageSpec to

>        determine whether to write an ASCII or binary file.

Author: probably deleted by mistake.

     - ptr
     - Pointer to a ``Filesystem::IOProxy`` that will handle the I/O, for
       example by reading from memory rather than the file system.
   * - ``pnm:bigendian``
     - int
     - If nonzero, the PNM file is big-endian (the default is little-endian).  
   * - ``pnm:pfmflip``
      - int
      - If nonzero, the PFM (float) file is flipped upside-down (the default

>       is flipped).

Reviewer: This is a little confusing. Is the default to flip or not to flip?

Author: Flipped. Because of GIMP and NetPBM that have this by default flipped. As result libjxl tools also export flipped pfm.

When I personally prefer non flipped

>       - If nonzero, the PFM (float) file is flipped upside-down (the default
>       is flipped).

Reviewer: ```suggestion
      - If this configuration hint is present and is zero, the automatic vertical flipping of PFM image will be disabled (i.e., scanline 0 will really be the first one stored in the file). If nonzero (the default), float PFM files will store scanline 0 as the last scanline in the file (i.e. the visual "top" of the image).
```

     - ptr
     - Pointer to a ``Filesystem::IOProxy`` that will handle the I/O, for
       example by writing to a memory buffer.
   * - ``pnm:bigendian``
     - int
     - If nonzero, the PNM file is big-endian (the default is little-endian).
   * - ``pnm:binary``
     - int
     - nonzero if the file itself used the PNM binary format, 0 if it used
       ASCII.  The PNM writer honors this attribute in the ImageSpec to
       determine whether to write an ASCII or binary file.
       Float PFM files are always written in binary format.
   * - ``pnm:pfmflip``
      - int

>       - If nonzero, the PFM (float) file is flipped upside-down (the default
>       is flipped).

Reviewer: ```suggestion
      - If this configuration hint is present and is zero, for PFM files, scanline 0 will really be stored first in the file, thus disabling the usual automatically flipping that accounts for PFM files conventionally being stored in bottom-to-top order. If nonzero (the default), float PFM files will store scanline 0 as the last scanline in the file (i.e. the visual "top" of the image).
```
